components/movementsensor/imuvectornav/imu.go

```diff
@@ -191,34 +192,29 @@ func newVectorNav(
 	if err != nil {
 		return nil, errors.Wrap(err, "couldn't configure deltaV register")
 	}
+
 	// tare the heading
 	err = v.vectornavTareSPI(ctx)
 	if err != nil {
 		return nil, err
 	}
-	smp := uint(100)
-	if v.polling > 0 {
-		smp = uint(v.polling)
-	}
-
 	// compensate for acceleration bias due to misalignement
-	err = v.compensateAccelBias(ctx, smp)
+	err = v.compensateAccelBias(ctx, v.polling)
```

Comment:
Hmm, a potential race with constructing and closing with a bad configuration is possible now since we're using the struct's `v.polling` rather than a new variable that's set to the value. However, this was more of a concern when we used to read and write spi directly to memory, which will require re-implementing this driver entirely if we go back to using that as an underlying mechanism for the imu, I don't know how worried to be about this.

Also could we do 
```suggestion
	if err = v.compensateAccelBias(ctx, v.polling); err != nil {
```
for brevity?

Reply:
> Hmm, a potential race [...] is possible now since we're using the struct's v.polling rather than a new variable that's set to the value. 

If there is a possibility of a data race since we are now using v.polling lets just go back to using the new variable instead.